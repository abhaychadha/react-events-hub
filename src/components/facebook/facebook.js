import React from "react";

const Facebook = props => {

  return (
    <a size="small" color="primary" 
      target="_blank" 
      href={props.social} 
      className= {props.classes.faceboolLinkContainer} 
      rel="noopener noreferrer">        
    </a>
  );
};

export default Facebook;
