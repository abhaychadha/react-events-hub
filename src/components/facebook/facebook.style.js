import FacebookIco from "../../static/facebook.png";
export const facebookLink = {
  faceboolLinkContainer: {
    minWidth: "20px",
    minHeight: "20px",
    background: `url(${FacebookIco})`,
    backgroundPosition: "center",
    backgroundRepeat: "no-repeat",
    backgroundSize: "contain",
    float: "right"
  }
};
