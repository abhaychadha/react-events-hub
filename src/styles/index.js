import { injectGlobal } from "styled-components";
import common from "./common";
import fonts from "./fonts.css";
injectGlobal`  
  ${fonts}
  ${common}
`;
