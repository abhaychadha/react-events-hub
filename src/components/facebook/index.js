import Facebook from "./facebook";
import { withStyles } from '@material-ui/core/styles';
import {facebookLink} from './facebook.style';

export default withStyles(facebookLink)(Facebook);