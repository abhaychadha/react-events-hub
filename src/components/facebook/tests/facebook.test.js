import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import Facebook from '../facebook';
import {facebookLink} from '../facebook.style';
import { withStyles } from '@material-ui/core/styles';

const StyledFacebook = withStyles(facebookLink)(Facebook);
describe('StyledFacebook', () => {
  let shallow;

  beforeEach(() => {
    shallow = createShallow(<StyledFacebook/>);
  });

  it('facebook renders with style', () => {
    const wrapper = shallow(<StyledFacebook />);
    expect(wrapper).toMatchSnapshot();
  });
});
