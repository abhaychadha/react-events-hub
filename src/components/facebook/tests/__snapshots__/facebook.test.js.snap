// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledFacebook facebook renders with style 1`] = `
<Facebook
  classes={
    Object {
      "faceboolLinkContainer": "Facebook-faceboolLinkContainer-1",
    }
  }
/>
`;
